File: Wrap-up/Java/GcdOfTwoNumbers.java
import java.util.Scanner;

public class GcdOfTwoNumbers {
    public static void main(String... args){
        Scanner input = new Scanner(System.in);
        
        IO.println("Get The GCD of Two Numbers");
        
        IO.println("Enter First Number: ");
        int userFirstNumber = input.nextInt();
        
        IO.println();
        
        IO.println("Enter Second Number: ");
        int userSecondNumber = input.nextInt();
        
        IO.println();
        
        int largerNumber = 0;
        int smallerNumber = 0;
        
        if(userFirstNumber > userSecondNumber){
            largerNumber = userFirstNumber;
            smallerNumber = userSecondNumber;
            
            while(smallerNumber != 0){
                int remainder = largerNumber % smallerNumber;
                largerNumber = smallerNumber;
                smallerNumber = remainder;
            }
        }
        else if(userFirstNumber < userSecondNumber){
            largerNumber = userSecondNumber;
            smallerNumber = userFirstNumber;
            
            while(smallerNumber != 0){
                int remainder = largerNumber % smallerNumber;
                largerNumber = smallerNumber;
                smallerNumber = remainder;
            }
        }
        
        int greatestCommonDivisor = largerNumber;
        
        IO.println("The Greatest Common Divisor -> " + greatestCommonDivisor);
    }
}

File: Wrap-up/python/gcd_of_two_numbers.py
print("Get The GCD of Two Numbers")

userFirstNumber = int(input("Enter First Number: "))

print()

userSecondNumber = int(input("Enter Second Number: "))

print()

largerNumber = 0
smallerNumber = 0

if(userFirstNumber > userSecondNumber):
    largerNumber = userFirstNumber
    smallerNumber = userSecondNumber
    
    while(smallerNumber != 0):
        remainder = largerNumber % smallerNumber
        largerNumber = smallerNumber
        smallerNumber = remainder
    
elif(userFirstNumber < userSecondNumber):
    largerNumber = userSecondNumber
    smallerNumber = userFirstNumber
    
    while(smallerNumber != 0):
        remainder = largerNumber % smallerNumber
        largerNumber = smallerNumber
        smallerNumber = remainder
    

greatestCommonDivisor = largerNumber

print("The Greatest Common Divisor -> ", greatestCommonDivisor)
